### PR TITLE
ADLS Gen2 Source documentation

### PR DESCRIPTION
**Pull request by stepgerner**
As per https://jira.wandisco.com/browse/DOCU-1512

### CODE REVIEW

File: docs/configure-storage.md [hunk 1]
```diff
@@ -6,10 +6,10 @@ sidebar_label: Configure filesystems
 
 The first step in the process of migrating data is to configure your filesystems, which defines where data is being migrated from (the source) and to (the target).
 
-* Supported sources are: **HDFS**, **Amazon Simple Storage Service (Amazon S3)** and **local storage**
+* Supported sources are: **ADLS Gen2 (preview)**, **HDFS**, **Amazon Simple Storage Service (Amazon S3)** and **local storage**
```
Comment (AshVic) [inline docs/configure-storage.md:9]: write product name in full first time followed by parentheses.

File: docs/configure-storage.md [hunk 2]
```diff
@@ -49,6 +50,44 @@ If you want to manually configure a source for LiveData Migrator to use, you mus
 If you have deleted the automatically discovered HDFS source but want to restore it, you can run `service livedata-migrator restart`. Upon restarting, LiveData Migrator will automatically attempt to discover the HDFS source again.
```
Comment (AshVic) [inline docs/configure-storage.md:50]: Think in the docs we've used "detected" but in the UI texts, we have "discovered"... I know it isn't relevant to this PR but we'll need to align the docs as you have here to use the term "discovery".

Comment (AshVic) [reply docs/configure-storage.md:50]: "Upon" restarting is too formal. 
"When LiveData Migrator restarts, it automatically attempts to discover the HDFS source again."

File: docs/configure-storage.md [hunk 3]
```diff
@@ -49,6 +50,44 @@ If you want to manually configure a source for LiveData Migrator to use, you mus
...
 :::
 
+#### Source ADLS Gen2 configuration (preview)
+
+:::note
+The creation of an Azure Data Lake Storage Gen2 filesystem as a source is currently a preview feature, and is subject to change.
```
Comment (AshVic) [inline docs/configure-storage.md:56]: Use the acronym here.

File: docs/configure-storage.md [hunk 4]
```diff
@@ -49,6 +50,44 @@ If you want to manually configure a source for LiveData Migrator to use, you mus
...
+
+At this time, ADLS Gen2 source filesystems you create can only be used for [one-time migrations](./one-time-migration.md).
+:::
+
+In the Filesystem panel, select to configure your _Unknown source_ and provide your source ADLS Gen2 configuration:
```
Comment (AshVic) [inline docs/configure-storage.md:61]: Bold for UI texts - Filesystem and Unknown source

Comment (AshVic) [reply docs/configure-storage.md:61]: There's a noun missing after "select".

File: docs/configure-storage.md [hunk 5]
```diff
@@ -49,6 +50,44 @@ If you want to manually configure a source for LiveData Migrator to use, you mus
...
+
+* **Filesystem Type** - The type of filesystem source. Choose _ADLS Gen2_.
```
Comment (AshVic) [inline docs/configure-storage.md:63]: Bold ADLS Gen2

File: docs/configure-storage.md [hunk 6]
```diff
@@ -49,6 +50,44 @@ If you want to manually configure a source for LiveData Migrator to use, you mus
...
+* **Display Name** - Provide a name for your source filesystem.
```
Comment (AshVic) [inline docs/configure-storage.md:64]: Enter

https://docs.microsoft.com/en-us/style-guide/procedures-instructions/describing-interactions-with-ui

File: docs/configure-storage.md [hunk 7]
```diff
@@ -49,6 +50,44 @@ If you want to manually configure a source for LiveData Migrator to use, you mus
...
+* **Authentication Type** - The authentication type to use when connecting to your filesystem. Choose either _Shared Key_ or _Service Principal (OAuth2)_.
```
Comment (AshVic) [inline docs/configure-storage.md:65]: Bold the UI texts

File: docs/configure-storage.md [hunk 8]
```diff
@@ -49,6 +50,44 @@ If you want to manually configure a source for LiveData Migrator to use, you mus
...
+* You will be asked to provide the security details of your Azure storage account. These will vary depending on which **Authentication Type** you choose. [See below](#shared-key).
+* **Use Secure Protocol** - This checkbox determines whether to use TLS encryption in communication with ADLS Gen2. This is enabled by default.
+
+:::note
+As ADLS Gen2 source filesystems can only be used for one-time migrations, the **Migrate live events** checkbox is unchecked and cannot be changed.
```
Comment (AshVic) [inline docs/configure-storage.md:70]: Use contractions where possible for less formal tone: can't

File: docs/configure-storage.md [hunk 9]
```diff
@@ -49,6 +50,44 @@ If you want to manually configure a source for LiveData Migrator to use, you mus
...
+:::
+
+The Azure storage account details necessary will vary depending on whether you selected _Shared Key_ or _Service Principal (OAuth2)_:
```
Comment (AshVic) [inline docs/configure-storage.md:73]: Use present tense "details vary depending on.."
Bold UI texts

File: docs/configure-storage.md [hunk 10]
```diff
@@ -49,6 +50,44 @@ If you want to manually configure a source for LiveData Migrator to use, you mus
...
+* **Access Key** - The access key associated with the Microsoft Azure account.
+* **Container Name** - The ADLS Gen2 container you want to migrate data from.
+
+##### Service Principal (OAuth2)
+
+* **Account Name** - The Microsoft Azure account name that owns the data lake storage.
+* **Container Name** - The ADLS Gen2 container you want to migrate data from.
+* **Client ID** - The client ID (also known as application ID) for your Azure service principal.
+* **Secret** - The client secret (also known as application secret) for the Azure service principal.
+* **Endpoint** - The client endpoint for the Azure service principal. This is referenced in the UI as Endpoint. This will often take the form of https://login.microsoftonline.com/{tenant}/oauth2/v2.0/token where {tenant} is the directory ID for the Azure service principal. You can specify a custom URL if desired (such as a proxy endpoint that manually interfaces with Azure Active Directory).
```
Comment (AshVic) [inline docs/configure-storage.md:87]: Bold UI text endpoint.

Use enter for consistency not "specify".

File: docs/configure-storage.md [hunk 11]
```diff
@@ -49,6 +50,44 @@ If you want to manually configure a source for LiveData Migrator to use, you mus
...
+
+Press **Save** once you've entered all necessary configuration details.
```
Comment (AshVic) [inline docs/configure-storage.md:89]: Select not "press" 

https://docs.microsoft.com/en-us/style-guide/procedures-instructions/describing-interactions-with-ui

File: docs/configure-storage.md [hunk 12]
```diff
@@ -57,8 +96,8 @@ Configure your source filesystem if Kerberos is enabled or Hadoop configuration
 
 In the Filesystem panel, select to configure your _Unknown source_ and provide your source HDFS configuration:
```
Comment (AshVic) [inline docs/configure-storage.md:97]: Bold UI texts pls
Also a noun is missing

File: docs/configure-storage.md [hunk 13]
```diff
@@ -57,8 +96,8 @@ Configure your source filesystem if Kerberos is enabled or Hadoop configuration
...
 
-* **Filesystem ID** - Provide a name for your source filesystem.
 * **Filesystem Type** - The type of filesystem source. Choose _HDFS_.
+* **Display Name** - Provide a name for your source filesystem.
```
Comment (AshVic) [inline docs/configure-storage.md:100]: Enter

https://docs.microsoft.com/en-us/style-guide/procedures-instructions/describing-interactions-with-ui

File: docs/configure-storage.md [hunk 14]
```diff
@@ -57,8 +96,8 @@ Configure your source filesystem if Kerberos is enabled or Hadoop configuration
 
 In the Filesystem panel, select to configure your _Unknown source_ and provide your source HDFS configuration:
 
-* **Filesystem ID** - Provide a name for your source filesystem.
```
Comment (AshVic) [inline docs/configure-storage.md:60]: Enter
https://docs.microsoft.com/en-us/style-guide/procedures-instructions/describing-interactions-with-ui

File: docs/configure-storage.md [hunk 15]
```diff
@@ -57,8 +96,8 @@ Configure your source filesystem if Kerberos is enabled or Hadoop configuration
...
 * **Filesystem Type** - The type of filesystem source. Choose _HDFS_.
+* **Display Name** - Provide a name for your source filesystem.
 * **Default FS** - Provide the `fs.defaultFS` value from your HDFS configuration.
 * Kerberos Configuration
   * **Kerberos Principal** - Provide a principal that will map to the [HDFS super user](https://hadoop.apache.org/docs/current/hadoop-project-dist/hadoop-hdfs/HdfsPermissionsGuide.html#The_Super-User) using [`auth_to_local`](https://hadoop.apache.org/docs/stable/hadoop-project-dist/hadoop-common/SecureMode.html#Mapping_from_Kerberos_principals_to_OS_user_accounts) rules, or add the LiveData Migrator user principal to the super-user group on the Hadoop cluster you're using.
```
Comment (AshVic) [inline docs/configure-storage.md:103]: **Enter** a,,,that maps (present tense)

File: docs/configure-storage.md [hunk 16]
```diff
@@ -167,9 +206,11 @@ Delete a target filesystem from its **Filesystem Configuration** panel. Click th
 
 ## Configure filesystems with the CLI
 
-### Validate your source
+LiveData Migrator migrates data from a single source filesystem. By default, LiveData Migrator will discover the HDFS source filesystem it's installed on. If automatic discovery fails, you can [validate the source](#validate-your-source-filesystem). As an alternative to the automatically discovered HDFS source filesystem, you can also [manually add a source filesystem of your required type](./add-file-systems) to use instead.
```
Comment (AshVic) [inline docs/configure-storage.md:209]: present tense "discovers"

Remove the term "fail" and make sentence more concise:

If LDM can't discover the HDFS source filesystem automatically, you can validate your source filesystem. Alternatively, you can add a source filesystem manually.

File: docs/configure-storage.md [hunk 17]
```diff
@@ -167,9 +206,11 @@ Delete a target filesystem from its **Filesystem Configuration** panel. Click th
...
+
+### Validate your source filesystem
 
-LiveData Migrator migrates data from a source filesystem. Verify that the correct source filesystem is registered or delete the existing one (you'll define a new source in the [Add File Systems](#add-file-systems) step).
+Verify that the correct source filesystem is registered or delete the existing one (you'll define a new source in the [Add File Systems](#add-file-systems) step).
```
Comment (AshVic) [inline docs/configure-storage.md:213]: Oxford comma before "or".

File: docs/configure-storage.md [hunk 18]
```diff
@@ -167,9 +206,11 @@ Delete a target filesystem from its **Filesystem Configuration** panel. Click th
...
 
 If Kerberos is enabled or your Hadoop configuration does not contain the information needed to connect to the Hadoop file system, use the [`filesystem auto-discover-source hdfs`](./command-reference.md#filesystem-auto-discover-source-hdfs) command to provide your Kerberos credentials and auto-discover your source HDFS configuration.
```
Comment (AshVic) [inline docs/configure-storage.md:215]: Contract where possible for less formal, conversational tone to "doesn't" "can't".

"Filesystem" should be one word?

"to **enter** your Kerberos credentials..."

File: docs/configure-storage.md [hunk 19]
```diff
@@ -192,13 +233,15 @@ Add file systems to provide LiveData Migrator with the information needed to rea
 
 A range of different file system types are supported as targets, including [ADLS Gen 2](https://docs.microsoft.com/en-us/azure/storage/blobs/data-lake-storage-introduction), [HDFS](https://hadoop.apache.org/docs/current/hadoop-project-dist/hadoop-hdfs/HdfsDesign.html), [GCS](https://cloud.google.com/storage), and [S3A](https://hadoop.apache.org/docs/current/hadoop-aws/tools/hadoop-aws/index.html).
```
Comment (AshVic) [inline docs/configure-storage.md:234]: Filesystems is one word.

File: docs/configure-storage.md [hunk 20]
```diff
@@ -192,13 +233,15 @@ Add file systems to provide LiveData Migrator with the information needed to rea
...
 
+LiveData Migrator supports [ADLS Gen 2](https://docs.microsoft.com/en-us/azure/storage/blobs/data-lake-storage-introduction), [HDFS](https://hadoop.apache.org/docs/current/hadoop-project-dist/hadoop-hdfs/HdfsDesign.html), [S3A](https://hadoop.apache.org/docs/current/hadoop-aws/tools/hadoop-aws/index.html) and a local filesystem as source filesystems.
```
Comment (AshVic) [inline docs/configure-storage.md:236]: Oxofrd comma before and.

File: docs/configure-storage.md [hunk 21]
```diff
@@ -192,13 +233,15 @@ Add file systems to provide LiveData Migrator with the information needed to rea
...
+
 :::note
-LiveData Migrator currently supports HDFS as a migration source.
+The creation of an Azure Data Lake Storage Gen2 filesystem as a source is currently a preview feature, and is subject to change.
```
Comment (AshVic) [inline docs/configure-storage.md:239]: Use the acronym here

File: docs/configure-storage.md [hunk 22]
```diff
@@ -192,13 +233,15 @@ Add file systems to provide LiveData Migrator with the information needed to rea
...
 
-If your source file system was not discovered automatically or you wish to assign a new source file system, use the `--source` parameter with the `filesystem add hdfs` command to add a suitable HDFS source file system.
+At this time, ADLS Gen2 source filesystems can only be used for [one-time migrations](./one-time-migration.md).
 :::
 
-You can define multiple target file systems, which you can migrate to at the same time.
+You can define multiple target file systems, which you can migrate to at the same time, but you can only define one source filesystem at a time.
```
Comment (AshVic) [inline docs/configure-storage.md:244]: Filesystems is one word?

File: docs/configure-storage.md [hunk 23]
```diff
@@ -250,12 +299,12 @@ LiveData Migrator will only perform *read* tasks on a source filesystem created
...
 
-```text="Code"
+```text title="Example"
 filesystem add hdfs --source --scanOnly ...
 ```
 
-:::note
-The account used to connect to a source filesystem intended for one-time migrations only requires read access. Write access is not necessary.
+:::info
+The account used to connect to a source filesystem intended for one-time migrations only requires read access to the filesystem. Write access is not necessary.
```
Comment (AshVic) [inline docs/configure-storage.md:307]: Contract to isn't?

File: docs/create-path-mappings.md [hunk 24]
```diff
@@ -23,8 +23,8 @@ Using path mapping enables any migrated data (relating to managed Hive databases
 
 You can create path mappings through the UI. To access path mappings:
 
-* Click on the LiveData Migrator hostname under your "Products" list in the dashboard.
-* Under the "Configuration" menu, select "Path Mappings".
+* Click on your intended LiveData Migrator instance under your "Products" list in the dashboard.
```
Comment (AshVic) [inline docs/create-path-mappings.md:26]: Select the LiveData Migrator instance under your **Products** list in the dashboard.

File: docs/create-path-mappings.md [hunk 25]
```diff
@@ -23,8 +23,8 @@ Using path mapping enables any migrated data (relating to managed Hive databases
 
 You can create path mappings through the UI. To access path mappings:
 
-* Click on the LiveData Migrator hostname under your "Products" list in the dashboard.
-* Under the "Configuration" menu, select "Path Mappings".
+* Click on your intended LiveData Migrator instance under your "Products" list in the dashboard.
+* Under the **Migrations** menu, select **Path Mappings**.
 
 Here, you will see a list of any existing path mappings.
```
Comment (AshVic) [inline docs/create-path-mappings.md:29]: Present tense is fine.

File: docs/prereqs.md [hunk 26]
```diff
@@ -6,7 +6,7 @@ sidebar_label: Prerequisites
 
 Ready to start? Follow these steps to get ready to use LiveData Migrator.
 
-[Read the release notes](https://community.wandisco.com/s/article/Release-Notes-WANdisco-LiveData-Migrator-Release-Notes-1-15-1) to get the latest information about the current version of LiveData Migrator.
+[Read the release notes](https://community.wandisco.com/s/article/Release-Notes-WANdisco-LiveData-Migrator-Release-Notes-1-16) to get the latest information about the current version of LiveData Migrator.
```
Comment (AshVic) [inline docs/prereqs.md:9]: Hyperlink on the nouns where possible ie **release notes**.
1.17?

File: docs/preview-features.md [hunk 27]
```diff
@@ -54,21 +54,22 @@ preview.feature.backup-restore=ON
 
 These features do not require enablement in the properties file. They are immediately available for use.
```
Comment (AshVic) [inline docs/preview-features.md:55]: don't
They're

File: docs/preview-features.md [hunk 28]
```diff
@@ -54,21 +54,22 @@ preview.feature.backup-restore=ON
...
 
-### S3 as a source filesystem
+### ADLS Gen2 as a source filesystem
 
-LiveData Migrator supports Amazon Simple Storage Service (Amazon S3) buckets as source filesystems.
+LiveData Migrator supports Azure Data Lake Storage Gen2 as source filesystems, but only for [one-time migrations](./one-time-migration.md) at this time.
```
Comment (AshVic) [inline docs/preview-features.md:59]: Use the acronym?
Should it be singular ? - as a source filesystem

File: docs/preview-features.md [hunk 29]
```diff
@@ -54,21 +54,22 @@ preview.feature.backup-restore=ON
...
+#### Creating an ADLS Gen2 source in the UI
 
-To configure an Amazon S3 bucket as a source filesystem, select **Amazon S3** in the **Filesystem Type** dropdown menu when [configuring filesystems with the UI](./configure-storage.md#configure-storage-with-the-ui).
+To configure an ADLS Gen2 system as a source filesystem, select **ADLS Gen2** in the **Filesystem Type** dropdown menu when [configuring filesystems with the UI](./configure-storage.md#configure-storage-with-the-ui).
 
-#### Creating an Amazon S3 source through the CLI
+#### Creating an ADLS Gen2 source through the CLI
 
-Supply the `--source` parameter to the [`filesystem add s3a` command](./command-reference.md#filesystem-add-s3a).
+Supply the `--source` parameter to the [`filesystem add adls2 oauth` or `filesystem add adls2 sharedkey` command](./command-reference.md#filesystem-add-adls2).
```
Comment (AshVic) [inline docs/preview-features.md:67]: For consistency in the style and patterning, I would just write "Enter the...parameter in the ...command."

File: docs/preview-features.md [hunk 30]
```diff
@@ -54,21 +54,22 @@ preview.feature.backup-restore=ON
...
 
 ```text title="Example"
-filesystem add s3a --file-system-id mytarget --bucket-name mybucket1 --credentials-provider org.apache.hadoop.fs.s3a.SimpleAWSCredentialsProvider --access-key B6ZAI18Z3UIO002Y777A --secret-key OP87Chokisf4hsTP0Q5j95yI904lT7AaDBGJpp0D --source
+filesystem add adls2 oauth --source --file-system-id mytarget --storage-account-name myadls2 --oauth2-client-id b67f67ex-ampl-e2eb-bd6d-client9385id --oauth2-client-secret 2IPO8*secretk-9OPs8n*TexampleHJ= --oauth2-client-endpoint https://login.microsoftonline.com/78u098ex-ampl-e498-8bce-ndpoint5f2e5/oauth2/v2.0/token --container-name lm2target
 ```
+
 ### Configure success files
 
 This feature is available for Hadoop Distributed File System (HDFS) source filesystems. Use success files to determine when a specific path has migrated successfully and the data within is ready for an application or job to process on the target side.
```
Comment (AshVic) [inline docs/preview-features.md:75]: Remove "within". It is very formal and redundant here. The data is ready.

**Review by AshVic — CHANGES_REQUESTED** [review]

File: docs/configure-smtp.md [hunk 31]
```diff
@@ -23,7 +23,11 @@ Configure an SMTP server in the UI by accessing the SMTP configuration tab and f
 ### SMTP Server Configuration Fields
 
 * **SMTP Email Server (Required)**: The host domain or IP address of your SMTP server.
-* **SMTP Connection Security (Required)**: The type of security your SMTP server uses. Can be **TLS** or **None**.
+* **SMTP Connection Security (Required)**: The type of security your SMTP server uses. Can be **TLS**, **StartTLS Enabled & Required**, **StartTLS Enabled** or **None**.
+	* **TLS**: Immediately uses a secure SMTPS connection. If it fails, the connection is aborted.
```
[the remark this answers is not in the document]
Comment (stepgerner) [reply docs/configure-smtp.md:27]: Thanks Ashleigh! I'm going to go with "terminated" in place of "aborted".

File: docs/preview-features.md (hunk 28 above)
Comment (stepgerner) [reply docs/preview-features.md:59]: Correct re: filesystem

**Pull request merged**